### DISCUSSION TITLE
Jint Debugging - set next statement

### DISCUSSION THREAD

**Discussion by martinbu** [sebastienros/jint#1549] [Q&A]
Hi,

I am currently analyzing the Jint Debugging possibilities and found examples at [Jint.DebugAdapter](https://github.com/Jither/Jint.DebugAdapter) and [Jint.DebuggerExample](https://github.com/Jither/Jint.DebuggerExample).

What I could not found until now is the possibility to explicitly set the next statement where the Jint engine should continue. So if a breakpoint is hit, to set the next statement to a previous one or to a statement after the current location skipping the lines in between.

So the main question is: Does the Jint Engine support explicitly setting which statement should be executed next?

**Comment by lahma** [comment]
Jint currently doesn't support skipping statements until some condition AFAIK.  /cc @Jither

**Reply by Jither** [reply]
Nope - that would probably be more of an engine functionality than a debugger functionality - there's no way for the debugger to tell Jint to change the actual execution - but then, most runtimes don't allow that either - it makes optimization and scope handling quite a bit trickier.
